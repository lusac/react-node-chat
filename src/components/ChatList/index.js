import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { getChats } from '../../actions/chats'
import { getChat } from '../../actions/chat'

export class ChatList extends React.Component {
  componentDidMount() {
    this.props.getChats()
  }

  selectChat(id) {
    if (this.props.socket.socket) {
      this.props.getChat(id)
      this.props.socket.socket.emit('room', id)
    }
  }

  render() {
    return (
      <div className="chat-list">
        {this.props.chats.loading && <span>loading...</span>}
        {!this.props.chats.loading && this.props.chats.chats.map((room) => {
          return <span key={room.id} className="chat-list__item" onClick={this.selectChat.bind(this, room.id)}>{room.name}</span>
        })}
      </div>
    )
  }
}

ChatList.propTypes = {
  chats: PropTypes.object,
  getChats: PropTypes.func,
  getChat: PropTypes.func,
  socket: PropTypes.object
};

function mapStateToProps(state) {
  return ({
    chat: state.chat,
    chats: state.chats,
    socket: state.socket
  })
}

export default connect(mapStateToProps, {
  getChats,
  getChat
})(ChatList)
