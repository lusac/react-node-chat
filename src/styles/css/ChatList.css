.chat-list {
  grid-area: list;
  display: flex;
  flex-direction: column;
}
.chat-list__item {
  height: 50px;
  display: flex;
  cursor: pointer;
  align-items: center;
  justify-content: center;
  border-bottom: 1px solid #EEE;
}
.chat-list__item:hover {
  background: #EFEFEF;
}
.chat-list__item:first-child {
  border-top: 1px solid #EEE;
}

/*# sourceMappingURL=ChatList.css.map */
