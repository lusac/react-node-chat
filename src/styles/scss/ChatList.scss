.chat-list {
  grid-area: list;
  display: flex;
  flex-direction: column;

  &__item {
    height: 50px;
    display: flex;
    cursor: pointer;
    align-items: center;
    justify-content: center;
    border-bottom: 1px solid #EEE;

    &:hover {
      background: #EFEFEF;
    }

    &:first-child {
      border-top: 1px solid #EEE;
    }
  }
}