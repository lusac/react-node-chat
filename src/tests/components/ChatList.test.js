/* global it describe expect beforeEach */
import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
import ChatList from '../../components/ChatList'

configure({ adapter: new Adapter() })

describe('ChatList component', () => {
  describe('snapshot', () => {
    it('empty list', () => {
      const rooms = {}
      const renderedValue =  renderer.create(<rooms rooms={rooms} />).toJSON()
      expect(renderedValue).toMatchSnapshot();
    })

    it('with rooms', () => {
      const rooms = { 1:{id:1, name:'room 1'}, 2:{id:2, name:'room 2'}, 3:{id:3, name:'room 3'}}
      const renderedValue =  renderer.create(<ChatList rooms={rooms} />).toJSON()
      expect(renderedValue).toMatchSnapshot();
    })
  })

  describe('rendered', () => {
    describe('list loaded', () => {
      let wrapper, rooms = { 1:{id:1, name:'room 1'}, 2:{id:2, name:'room 2'}, 3:{id:3, name:'room 3'}}

      beforeEach(() => {
        wrapper = shallow(<ChatList rooms={rooms} />)
      })

      it('correct label', () => {
        expect(wrapper.find('span').get(0).props.children).toBe('room 1')
        expect(wrapper.find('span').get(1).props.children).toBe('room 2')
        expect(wrapper.find('span').get(2).props.children).toBe('room 3')
      })
    })
  })
})