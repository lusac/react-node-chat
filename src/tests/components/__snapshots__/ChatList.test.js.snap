// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList component snapshot empty list 1`] = `
<rooms
  rooms={Object {}}
/>
`;

exports[`ChatList component snapshot with rooms 1`] = `
<div
  className="chat-list"
>
  <span
    className="chat-list__item"
    onClick={[Function]}
  >
    room 1
  </span>
  <span
    className="chat-list__item"
    onClick={[Function]}
  >
    room 2
  </span>
  <span
    className="chat-list__item"
    onClick={[Function]}
  >
    room 3
  </span>
</div>
`;
